jaeger-dropwizard/src/main/java/com/uber/jaeger/dropwizard/JerseyServerFilter.java

```diff
@@ -0,0 +1,146 @@
+import com.google.common.annotations.VisibleForTesting;
+
+import com.uber.jaeger.context.TraceContext;
+import com.uber.jaeger.filters.jaxrs2.ServerFilter;
+import io.opentracing.Tracer;
+import org.glassfish.jersey.server.ContainerRequest;
+import org.glassfish.jersey.server.ExtendedUriInfo;
+import org.glassfish.jersey.server.internal.routing.UriRoutingContext;
+import org.glassfish.jersey.uri.UriTemplate;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import javax.ws.rs.Path;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.ext.Provider;
+import lombok.Value;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * A {@link ServerFilter} that provides a Jersey specific {@link #getOperationName} implementation.
+ * When using this Filter with Jersey, the full @Path template hierarchy is used as the
+ * operation name.
+ *
+ * WARNING: This class relies on an private implementation detail
+ * {@link UriTemplate#normalizedTemplate} to provide functionality.
+ */
+@Slf4j
+@Provider
+public class JerseyServerFilter extends ServerFilter {
+
+  private Map<CacheKey, String> methodToPathCache = new ConcurrentHashMap<>();
+  private Field normalizedTemplate;
+
+  public JerseyServerFilter(Tracer tracer,
+                            TraceContext traceContext) {
+    super(tracer, traceContext);
+
+    try {
+      normalizedTemplate = UriTemplate.class.getDeclaredField("normalizedTemplate");
+      normalizedTemplate.setAccessible(true);
+    } catch (NoSuchFieldException e) {
+      //TODO: Add metrics for failure counts
+      log.error("Cannot access the normalizedTemplate field from UriTemplate. "
+                    + "Path operation name for tracing will be disabled.", e);
+    }
+  }
+
+  /**
+   * Gets an operation name by first calling {@link #getPathOperationName(ContainerRequestContext)},
+   * and falling back on {@link #getResourceMethodOperationName(ContainerRequestContext)}.
+   * @return operation name
+   */
+  @Override
+  protected String getOperationName(ContainerRequestContext containerRequestContext) {
+    CacheKey key  = CacheKey.of(resourceInfo.getResourceMethod(), containerRequestContext.getMethod());
+    String operationName = methodToPathCache.get(key);
+    if (operationName != null) {
+      return operationName;
+    }
+
+    operationName = getPathOperationName(containerRequestContext);
+
+    if (operationName.isEmpty()) {
+      operationName = getResourceMethodOperationName(containerRequestContext);
+    }
+
+    methodToPathCache.put(key, operationName);
+    return operationName;
+  }
+
+  /**
+   * Retrieves an operation name that uses the resource method name of the matched jersey resource
+   * @return the operation name
+   */
+  private String getResourceMethodOperationName(ContainerRequestContext containerRequestContext) {
+    Method method = resourceInfo.getResourceMethod();
```

Comment:
here's what I don't understand. Where does the `resourceInfo` property come from? It looks like an inherited member variable, which implies there's going to be a new instance of this filter class for every method, if not for every http call, which makes the cache non-reusable.